cypress/integration/researchGroupCanSignUp.spec.js

```diff
@@ -0,0 +1,52 @@
+describe("Research Group Sign-Up", () => {
+  beforeEach(function() {
+    cy.server();
+    cy.route({
+      method: "GET",
+      url: "http://localhost:3000/api/v0/articles",
+      response: "fixture:articles.json",
+      status: 200
+    });
+  });
+  it("User can successfully sign-up as a Research Group w/ Keys", () => {
+    cy.route({
+      method: "POST",
+      url: "http://localhost:3000/api/v0/auth",
+      response: "fixture:successful_research_group_sign_up.json"
+    });
+    cy.research_group_successful_signup(
+      "research_group",
+      "Cancer Research Group",
+      "cancer_research@mail.com",
+      "password",
+      "password",
+      "abcd1234"
+    );
+    cy.get("#submit-account-button").click();
+    cy.get("#flash").should(
+      "contain",
+      "Your research_group account successfully created!"
+    );
+  });
+  it("User can't sign up without valid Subscription Key", () => {
+    cy.route({
+      method: "POST",
+      url: "http://localhost:3000/api/v0/auth",
+      response:"fixture:unsuccessful_research_group_sign_up.json",
+      status: 422
+    });
+    cy.research_group_successful_signup(
+      "research_group",
+      "Cancer Research Group",
+      "cancer_research@mail.com",
+      "password",
+      "password",
+      "wrong_registration_key"
+    );
+    cy.get("#submit-account-button").click();
+    cy.get("#flash").should(
+      "contain",
+      "Invalid registration key. Please try again."
```

Comment:
I was expecting to see some logic on the front end to initially check the validity of the key before it got sent back to the backend, as described in the MVC diagram idea: https://www.pivotaltracker.com/n/projects/2384164/stories/167917123

This may not be needed for minimum functionality but I bring it up to make sure there is a deliberate decision to be made about this security tradeoff.

Reply:
This would be good for added security, I agree.  But we are talking about MVP here and we need to get as much done tonight as possible.  Maybe we could make a chore to do this after we deliver MVP tomorrow?  I feel like that is a good compromise.